ext/date/php_date.c

```diff
@@ -4357,7 +4357,11 @@ static int php_date_interval_initialize_from_hash(zval **return_value, php_inter
 		zval *z_arg = zend_hash_str_find(myht, element, sizeof(element) - 1); \
 		if (z_arg && Z_TYPE_P(z_arg) <= IS_STRING) { \
 			zend_string *str = zval_get_string(z_arg); \
-			DATE_A64I((*intobj)->diff->member, ZSTR_VAL(str)); \
+			if (str->len) { \
```

Comment:
This should actually use the proper accessor macro, i.e.
```suggestion
			if (ZSTR_LEN(str)) { \
```